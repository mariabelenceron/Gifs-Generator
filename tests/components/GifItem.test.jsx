import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Testing <GifItem />', () => {
    const title = 'Adventure Time';
    const url = 'https://github.com/';
    
    test('Verified the snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('show the img with the URL and ALT element', () => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should show the title in the component', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});