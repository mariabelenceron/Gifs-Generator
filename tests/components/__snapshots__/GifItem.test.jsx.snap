// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to <GifItem /> Verified the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Adventure Time"
      src="https://github.com/"
    />
    <p>
      Adventure Time
    </p>
  </div>
</div>
`;
